nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_08.py

```diff
@@ -64,22 +72,39 @@ def transform_instrument_slot(self, omics_doc: dict):
             instruments = load_yaml_asset("migrator_from_10_2_0_to_11_0_0_part_08/instrument_set.yaml")
             allowed_instrument_names = [instrument["name"] for instrument in instruments]
 
+            # Determine which of the allowed instrument names most closely matches the "existing" (original) one.
+            # Reference: https://docs.python.org/3/library/difflib.html#difflib.get_close_matches
             cutoff = 0.25
             matches = difflib.get_close_matches(existing_instrument_name, allowed_instrument_names, n=1, cutoff=cutoff)
 
-            if matches:
+            if len(matches) > 0:
+                allowed_instrument_name = matches[0]  # this is the most closely-matching name
 
-                    # Match instrument_name with instrument set and get corresponding instrument id
-                    match = matches[0]
-                    self.logger.info(f"instrument_name in {omics_doc['id']} is {omics_doc['instrument_name']} and will be matched to instrument with name {match}")
+                # SPECIAL CASE: If the name of the original instrument contains—but does not end with—"NovaSeq",
+                #               (e.g. "NovaSeq S4" or "NovaSeq SP"), map it to "Illumina NovaSeq 6000".
+                if "NovaSeq" in omics_doc['instrument_name'] and not omics_doc['instrument_name'].endswith("NovaSeq"):
+                    instrument = self.adapter.get_document_having_value_in_field(
+                        collection_name="instrument_set", field_name="name", value="Illumina NovaSeq 6000"
+                    )
+
+                    self.logger.info(f"'instrument_name' in '{omics_doc['id']}' is '{omics_doc['instrument_name']}' "
+                                     f"and will be mapped to instrument named 'Illumina NovaSeq 6000'")
```

Comment:
Since this snippet doesn't depend upon `matches`, I think I will move it outside of the `if matches` (now `if len(matches) > 0`) block.